File: src/com/wang/Game2dEngine/painter/interfaces/IEngineGraphics.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.wang.Game2dEngine.painter.interfaces;

import java.awt.*;
import java.awt.geom.AffineTransform;
import java.awt.image.BufferedImage;

/**
 * @author ricolwang
 */
public interface IEngineGraphics
{

    void drawString(String text, float x, float y);

    void drawImage(Image img, int x, int y, int width, int height);

    void setColor(Color color);

    void fillRect(int x, int y, int width, int height);

    void dispose();

    void drawRect(int x, int y, int width, int height);

    void drawLine(int x0, int y0, int x1, int y1);

    void fillArc(int x, int y, int width, int height, int startAngel, int endAngel);

    void drawArc(int x, int y, int width, int height, int startAngel, int endAngel);

    void setBackground(Color blackTransparent);

    void clearRect(int x, int y, int width, int height);

    AffineTransform getTransform();

    void rotate(double theta, float x, float y);

    void setTransform(AffineTransform t);

    Composite getComposite();

    void setComposite(AlphaComposite ac);

    void drawImage(BufferedImage image, int x, int y);

    void setComposite(Composite c);

    void setFont(Font f);

    void fillOval(int x, int y, int width, int height);

    void drawPolygon(Polygon p);

    void setStroke(BasicStroke basicStroke);
}

File: src/com/wang/Game2dEngine/painter/interfaces/IEngineGraphics.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.wang.Game2dEngine.painter.interfaces

import java.awt.*
import java.awt.geom.AffineTransform
import java.awt.image.BufferedImage

/**
 * @author ricolwang
 */
interface IEngineGraphics
{
    fun drawString(text: String?, x: Float, y: Float)
    fun drawImage(img: Image?, x: Int, y: Int, width: Int, height: Int)
    fun setColor(color: Color?)
    fun fillRect(x: Int, y: Int, width: Int, height: Int)
    fun dispose()
    fun drawRect(x: Int, y: Int, width: Int, height: Int)
    fun drawLine(x0: Int, y0: Int, x1: Int, y1: Int)
    fun fillArc(x: Int, y: Int, width: Int, height: Int, startAngel: Int, endAngel: Int)
    fun drawArc(x: Int, y: Int, width: Int, height: Int, startAngel: Int, endAngel: Int)
    fun setBackground(blackTransparent: Color?)
    fun clearRect(x: Int, y: Int, width: Int, height: Int)
    var transform: AffineTransform?
    fun rotate(theta: Double, x: Float, y: Float)
    fun getComposite(): Composite?
    fun setComposite(ac: AlphaComposite?)
    fun drawImage(image: BufferedImage?, x: Int, y: Int)
    fun setComposite(c: Composite?)
    fun setFont(f: Font?)
    fun fillOval(x: Int, y: Int, width: Int, height: Int)
    fun drawPolygon(p: Polygon?)
    fun setStroke(basicStroke: BasicStroke?)
}